reward/staking_manager.go

```diff
@@ -103,14 +103,29 @@ func GetStakingManager() *StakingManager {
 	return stakingManager
 }
 
-// GetStakingInfo returns a corresponding stakingInfo for a blockNum.
+// GetStakingInfo returns a stakingInfo on the staking block of the given block number.
+// Note that staking block is the block on which the associated staking information is stored and used during an interval.
 func GetStakingInfo(blockNum uint64) *StakingInfo {
+	stakingBlockNumber := params.CalcStakingBlockNumber(blockNum)
+	return GetStakingInfoOnStakingBlock(stakingBlockNumber)
+}
+
+// GetStakingInfoOnStakingBlock returns a corresponding StakingInfo for a staking block number.
+// If the given number is not on the staking block, it returns nil.
+func GetStakingInfoOnStakingBlock(stakingBlockNumber uint64) *StakingInfo {
 	if stakingManager == nil {
 		logger.Error("unable to GetStakingInfo", "err", ErrStakingManagerNotSet)
 		return nil
 	}
 
-	stakingBlockNumber := params.CalcStakingBlockNumber(blockNum)
+	return getStakingInfo(0, stakingBlockNumber)
+}
+
+func getStakingInfo(blockNum, stakingBlockNumber uint64) *StakingInfo {
```

Comment:
Can we remove the `blockNum` argument? It seems that `blockNum` argument is not used except logging.

Reply:
@blukat29 Ture, but it is used in `GetStakingInfo` as well and it may not be a 0. Do you think still it is not necessary?

Reply:
- As per your change,
  - `GetStakingInfo(n)` -> `GetStakingInfoOnStakingBlock(m)` -> `getStakingInfo(0, m)`
  - `GetStakingInfoOnStakingBlock(m)` -> `getStakingInfo(0,m)`
  - The private function `getStakingInfo` not used anywhere else.
- So I thought we can simplify to
  - `GetStakingInfo` -> `GetStakingInfoOnStakingBlock`
  - `GetStakingInfoOnStakingBlock`

- But if you expect more use cases of `getStakingInfo`, (especially with nonzero blockNum) then I am okay with this change.

- Regarding the blockNum,
  - blockNum=0 can be misleading in the logs. Isn't it supposed to be like blockNum=172877, staking block number=86400? logging blockNum=0 conveys no meaning. If my understanding about blockNum is incorrect, please tell me.
  - Or we can call `getStakingInfo(stakingBlockNumber, stakingBlockNumber)` for clarity

Reply:
@blukat29 I agreed with you. It would be misleading the logs and the same as `getStakingInfo(stakingBlockNumber, stakingBlockNumber)`. I added a line of log for the original block number when calling `GetStakingInfo` only. Please take another look.